dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AliasResource.java

```diff
@@ -0,0 +1,157 @@
+
+        final WorkflowVersion workflowVersion = this.workflowVersionDAO.findByAlias(alias);
+        if (workflowVersion == null) {
+            LOG.error("Could not find workflow version using the alias: " + alias);
+            throw new CustomWebApplicationException("Workflow version not found when searching with alias: " + alias, HttpStatus.SC_BAD_REQUEST);
+        }
+
+        long workflowVersionId = workflowVersion.getId();
+        long workflowId = getWorkflowId(workflowVersionId);
+        Workflow workflow = workflowDAO.findById(workflowId);
+        workflowResource.checkEntry(workflow);
+        workflowResource.optionalUserCheckEntry(user, workflow);
+
+        return new WorkflowVersion.WorkflowVersionPathInfo(workflow.getWorkflowPath(), workflowVersion.getName());
+    }
+
+    @Override
+    public Optional<PublicStateManager> getPublicStateManager() {
+        return Optional.empty();
+    }
+
+    @Override
+    public WorkflowVersion getAndCheckResource(User user, Long workflowVersionId) {
+        final WorkflowVersion workflowVersion = this.workflowVersionDAO.findById(workflowVersionId);
+        if (workflowVersion == null) {
+            LOG.error("Could not find workflow version using the workflow version id: " + workflowVersionId);
+            throw new CustomWebApplicationException("Workflow version not found when searching with id: " + workflowVersionId, HttpStatus.SC_BAD_REQUEST);
+        }
+
+        Long workflowId = getWorkflowId(workflowVersionId);
+        Workflow workflow = workflowDAO.findById(workflowId);
+        workflowResource.checkEntry(workflow);
+        workflowResource.checkUserCanUpdate(user, workflow);
+        return workflowVersion;
+    }
+
+    @Override
+    // TODO: EntryResource.java also throws an exception for this method; need good explanation for why
+    public WorkflowVersion getAndCheckResourceByAlias(String alias) {
+        throw new UnsupportedOperationException("Use the TRS API for tools and workflows");
+    }
+
+    @Override
+    public WorkflowVersion addAliasesAndCheck(User user, Long id, String aliases, boolean blockFormat) {
+        WorkflowVersion c = getAndCheckResource(user, id);
+        Set<String> oldAliases = c.getAliases().keySet();
+        Set<String> newAliases = Sets.newHashSet(Arrays.stream(aliases.split(",")).map(String::trim).toArray(String[]::new));
+
+        AliasableResourceInterface.checkAliases(newAliases, user, blockFormat);
+
+        Set<String> duplicateAliasesToAdd = Sets.intersection(newAliases, oldAliases);
+        if (!duplicateAliasesToAdd.isEmpty()) {
+            String dupAliasesString = String.join(", ", duplicateAliasesToAdd);
+            throw new CustomWebApplicationException("Aliases " + dupAliasesString + " already exist; please use unique aliases",
+                    HttpStatus.SC_BAD_REQUEST);
+        }
+
+        newAliases.forEach(alias -> c.getAliases().put(alias, new Alias()));
+        return c;
```

Comment:
This is nitpicky, but c is not a very readable name.

Reply:
fixed